### ISSUE TITLE
Feature request: support appending query string in redirect plugin

### ISSUE THREAD

**Issue by liuxiran** [apache/apisix-dashboard#2095]
# Feature request

## Please describe your feature

Dashboard has integrated the redirect plugin into the first step of Route creation, refer to https://github.com/apache/apisix/pull/4298, it would be better to add `append_query_string` field

**Comment by namangirdhar16** [comment]
Hi @liuxiran @bzp2010 i would like to work on this issue, please assign this to me.

**Comment by zaunist** [comment]
Hi, @namangirdhar16 .Did you have any difficulties?

**Comment by juzhiyuan** [comment]
@zaunist could you help to check this?

**Comment by zaunist** [comment]
> @zaunist could you help to check this?

A great pleasure

**Comment by zaunist** [comment]
I checked [redirect plugin](https://apisix.apache.org/docs/apisix/plugins/redirect/), `Only one of http_to_https, uri or regex_uri can be specified.`, but `append_query_string` not one of them. So `append_query_string` maybe a separate button or option menu, because `append_query_string` can coexist with `http_to_https, uri, regex_uri`. Is it right ?

**Comment by juzhiyuan** [comment]
> I checked [redirect plugin](https://apisix.apache.org/docs/apisix/plugins/redirect/), `Only one of http_to_https, uri or regex_uri can be specified.`, but `append_query_string` not one of them. So `append_query_string` maybe a separate button or option menu, because `append_query_string` can coexist with `http_to_https, uri, regex_uri`. Is it right ?

Yes, both append_query_string & encode_uri are two seprate buttons, we could use Switch component to handle this.

**Comment by bzp2010** [comment]
ref #2981
